python/pyarrow/memory.pxi

```diff
@@ -127,6 +127,44 @@ def logging_memory_pool(MemoryPool parent):
     return out
 
 
+def system_memory_pool():
+    """
+    Return a memory pool based on the C malloc heap.
+    """
+    cdef:
+        MemoryPool pool = MemoryPool.__new__(MemoryPool)
+    pool.init(c_system_memory_pool())
+    return pool
+
+
+def jemalloc_memory_pool():
+    """
+    Return a memory pool based on the jemalloc heap.
+
+    NotImplemented is raised if this Arrow build does not enable jemalloc.
```

Comment:
Perhaps you should add a bit more context about where we use the allocators. Like jemalloc is used and enabled in our packages.

Reply:
I don't think jemalloc is enabled on all platforms, is it?

Reply:
*unix packages

Reply:
Well, we'll certain forgot to update this docstring if we change the build settings, so I'd rather not be more precise.

Reply:
Right